File: python/fibonacciNumber/solution2.py
storedFibs = {}
class Solution:
    def fib(self, N: int) -> int:
        if N in storedFibs:
            return storedFibs[N]
        elif N <= 1:
            result = N
        else:
            result = self.fib(N-1)+self.fib(N-2)
        storedFibs[N] = result
        return result 

File: Javascript/fibonacciNumber/solution2.js
let storedFibs = {};
// using memoization
var fib = function(N) {
    let result;
    if(storedFibs[N]){
        return storedFibs[N];
    } else if(N <=1){
        result = N;
    }else {
        result = fib(N-1)+fib(N-2);
    } 
    storedFibs[N] = result;
    return result;
    
};